src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,794 @@
+  const [unsavedChanges, setUnsavedChanges] = useState(false);
+
+  /** @type {[ApplicationDetails, React.Dispatch<React.SetStateAction<ApplicationDetails>>]} */
+  const [applicationDetails, setApplicationDetails] = useState({
+    ApplicationID: null,
+    DateSubmitted: null,
+    DateModified: null,
+    EditorUsername: null,
+    EditorEmail: null,
+    Applicants: [],
+    ApartmentChoices: [],
+  });
+  /** @type {[Number, React.Dispatch<React.SetStateAction<Number>>]} */
+  const [applicationID, setApplicationID] = useState(null); // Default value of -1 indicate to backend that the application ID number is not yet known
+  const [dateSubmitted, setDateSubmitted] = useState(null); // The date the application was submitted, or null if not yet submitted
+  const [dateModified, setDateModified] = useState(null); // The date the application was last modified, or null if not yet saved/modified
+  const [editorUsername, setEditorUsername] = useState(null); // The username of the application editor
+  /** @type {[ApartmentApplicant[], React.Dispatch<React.SetStateAction<ApartmentApplicant[]>>]} Array of applicant info */
+  const [applicants, setApplicants] = useState([]);
+  /** @type {[ApartmentChoice[], React.Dispatch<React.SetStateAction<ApartmentChoice[]>>]} Array of apartment choice info */
+  const [preferredHalls, setPreferredHalls] = useState([]); // Properties 'HallName' and 'HallRank' must be capitalized to match the backend
+
+  const [agreements, setAgreements] = useState(false); // Represents the state of the agreements card. True if all checkboxes checked, false otherwise
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('info');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  /**
+   * Loads the user's saved apartment application, if one exists
+   */
+  useEffect(() => {
+    let isSubscribed = true;
+
+    const handleError = (_error) => {
+      // No existing application was found in the database,
+      // or an error occurred while attempting to load the application
+      setApplicationID(null);
+      setEditorUsername((prevEditorUsername) => prevEditorUsername ?? userProfile.AD_Username);
+      setApplicants((prevApplicants) =>
+        prevApplicants.length ? prevApplicants : [{ Profile: userProfile, OffCampusProgram: '' }],
+      );
+      setUnsavedChanges(true);
+    };
+
+    if (!applicationID || !applicationDetails) {
```

Comment:
This check is not necessary, since it will always be true when we first load.
```suggestion
```

Reply:
That is how it _should_ be, but in reality, it often attempts to run the useEffect at least twice at first load, causing many of the other problems. This check was the only way I could figure out how to force it to not attempt the loading process multiple times, and as you commented above it is clearly not working correctly